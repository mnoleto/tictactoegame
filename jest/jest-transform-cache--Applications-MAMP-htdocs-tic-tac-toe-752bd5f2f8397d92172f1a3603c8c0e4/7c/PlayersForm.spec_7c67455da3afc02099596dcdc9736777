/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_PlayersForm = require('./PlayersForm'); /* istanbul ignore next */var _PlayersForm2 = _interopRequireDefault(_PlayersForm);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayersForm is responsible for render the game grid
// CONTRACTS
// 1. 

describe('PlayersForm', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayersForm = /* istanbul ignore next */void 0;
  var playersForm = function playersForm() {
    if (!mountedPlayersForm) {
      mountedPlayersForm = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));

    }
    return mountedPlayersForm;
  };

  beforeEach(function () {
    props = {};
    mountedPlayersForm = undefined;
  });

  it('render the PlayersForm', function () {
    var playersFormWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, /* istanbul ignore next */null));
    expect(playersFormWrapper).toMatchSnapshot();
  });

  it('render a div as root element', function () {
    var playersFormElement = playersForm().find('form');
    var wrappingDiv = playersFormElement.first();
    expect(wrappingDiv.children()).toEqual(playersForm().children());
  });
});