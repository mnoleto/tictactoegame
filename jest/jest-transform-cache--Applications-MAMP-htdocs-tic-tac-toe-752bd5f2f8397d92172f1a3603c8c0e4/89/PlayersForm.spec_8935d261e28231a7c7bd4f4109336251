/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_PlayersForm = require('./PlayersForm'); /* istanbul ignore next */var _PlayersForm2 = _interopRequireDefault(_PlayersForm);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayersForm is responsible for render the game grid
describe('PlayersForm', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayersForm = /* istanbul ignore next */void 0;
  var playersForm = function playersForm() {
    if (!mountedPlayersForm) {
      mountedPlayersForm = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));

    }
    return mountedPlayersForm;
  };

  beforeEach(function () {
    props = {
      onStartGame: jest.fn(),
      players: [],
      result: { status: 'waiting' },
      turn: '' };

    mountedPlayersForm = undefined;
  });

  it('should render the PlayersForm', function () {
    var playersFormWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));
    expect(playersFormWrapper).toMatchSnapshot();
  });

  it('should render a form as root element', function () {
    var playersFormElement = playersForm().find('form');
    var wrapping = playersFormElement.first();
    expect(wrapping.children()).toEqual(playersForm().children());
  });

  describe('legend', function () {
    it('should render a legend inside the form if the result.status is not running', function () {
      var playersFormRenderer = playersForm();
      var legendElement = playersForm().find('legend');

      playersFormRenderer.setProps({ result: { status: 'waiting' } });
      expect(legendElement).toBeDefined();
    });

    it('should not render a legend inside the form if the result.status is running', function () {
      var playersFormRenderer = playersForm();

      playersFormRenderer.setProps({ result: { status: 'running' } });
      var legendElement = playersForm().find('legend');
      expect(legendElement.length).toEqual(0);
    });
  });

  describe('div.player-input', function () {
    it('should render two div.render-input', function () {

    });
  });

  describe('start game button', function () {
    it('should render the button.start-game-button inside the form if the result.status is not running', function () {

    });

    it('should not render the button.start-game-button inside the form if the result.status is running', function () {

    });

    it('simulate click event', function () {

    });
  });
});