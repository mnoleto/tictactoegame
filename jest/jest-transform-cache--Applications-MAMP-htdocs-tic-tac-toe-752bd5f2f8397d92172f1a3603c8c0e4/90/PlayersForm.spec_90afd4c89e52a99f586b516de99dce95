/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_sinon = require('sinon'); /* istanbul ignore next */var _sinon2 = _interopRequireDefault(_sinon);
var /* istanbul ignore next */_PlayersForm = require('./PlayersForm'); /* istanbul ignore next */var _PlayersForm2 = _interopRequireDefault(_PlayersForm);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayersForm is responsible for render the players form
describe('<PlayersForm />', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayersForm = /* istanbul ignore next */void 0;
  var playersForm = function playersForm() {
    if (!mountedPlayersForm) {
      mountedPlayersForm = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));

    }
    return mountedPlayersForm;
  };

  beforeEach(function () {
    props = {
      onStartGame: jest.fn(),
      players: [],
      result: { status: 'waiting' },
      turn: '' };

    mountedPlayersForm = undefined;
  });

  it('should render the PlayersForm', function () {
    var playersFormWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));
    expect(playersFormWrapper).toMatchSnapshot();
  });

  it('form contains everything else that gets rendered', function () {
    var playersFormElement = playersForm().find('form');
    var wrapping = playersFormElement.first();
    expect(wrapping.hasClass('players-form')).toBeTruthy();
    expect(wrapping.children()).toEqual(playersForm().children());
  });

  describe('legend', function () {
    it('should render a legend inside the form if the result.status is not running', function () {
      var playersFormRenderer = playersForm();
      playersFormRenderer.setProps({ result: { status: 'waiting' } });
      var legendElement = playersForm().find('legend');
      expect(legendElement.length).toEqual(1);
    });

    it('should not render a legend inside the form if the result.status is running', function () {
      var playersFormRenderer = playersForm();
      playersFormRenderer.setProps({ result: { status: 'running' } });
      var legendElement = playersForm().find('legend');
      expect(legendElement.length).toEqual(0);
    });
  });

  describe('div.player-input', function () {
    it('should render two div.player-input', function () {
      var inputElements = playersForm().find('div.player-input');
      expect(inputElements.length).toEqual(2);
    });
  });

  describe('start game button', function () {
    it('should render the button.start-game-button inside the form if the result.status is not running', function () {
      var playersFormRenderer = playersForm();
      playersFormRenderer.setProps({ result: { status: 'waiting' } });
      var buttonElement = playersForm().find('button');
      expect(buttonElement.length).toEqual(1);
    });

    it('should not render the button.start-game-button inside the form if the result.status is running', function () {
      var playersFormRenderer = playersForm();
      playersFormRenderer.setProps({ result: { status: 'running' } });
      var buttonElement = playersForm().find('button');
      expect(buttonElement.length).toEqual(0);
    });

    it('simulate click event', function () {
      props.onStartGame = /* istanbul ignore next */_sinon2.default.spy();
      var buttonElement = playersForm().find('button');

      buttonElement.simulate('click');
      expect(props.onStartGame.callCount).toEqual(1);
    });
  });

  describe('Test methods', function () {
    var playersFormRenderer = /* istanbul ignore next */void 0;
    beforeEach(function () {
      playersFormRenderer = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));
    });

    describe('componentWillReceiveProps', function () {
      it('should change the players state', function () {
        playersFormRenderer.setProps({ players: ['Player 1', 'Player 2'] });
        playersFormRenderer.instance().componentWillReceiveProps();
        expect(playersFormRenderer.state().playerX).toEqual('Player 1');
        expect(playersFormRenderer.state().playerO).toEqual('Player 2');
      });
    });

    describe('handleInputChange', function () {
      it('should change the state', function () {
        playersFormRenderer.instance().handleInputChange('playerX', 'John');
        expect(playersFormRenderer.state().playerX).toEqual('John');

        playersFormRenderer.instance().handleInputChange('playerX', 'Mark');
        expect(playersFormRenderer.state().playerX).toEqual('Mark');

        playersFormRenderer.instance().handleInputChange('playerO', 'Player 3');
        expect(playersFormRenderer.state().playerO).toEqual('Player 3');
      });
    });

    describe('sendData', function () {
      it('should call onStartGame', function () {
        props.onStartGame = /* istanbul ignore next */_sinon2.default.spy();
        playersFormRenderer.instance().sendData();
        expect(props.onStartGame.callCount).toEqual(1);
      });
    });
  });


});