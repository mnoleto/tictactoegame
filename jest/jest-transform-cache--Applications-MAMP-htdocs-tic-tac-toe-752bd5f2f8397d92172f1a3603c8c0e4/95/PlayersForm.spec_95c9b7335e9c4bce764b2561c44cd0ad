/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_PlayersForm = require('./PlayersForm'); /* istanbul ignore next */var _PlayersForm2 = _interopRequireDefault(_PlayersForm);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayersForm is responsible for render the game grid
describe('PlayersForm', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayersForm = /* istanbul ignore next */void 0;
  var playersForm = function playersForm() {
    if (!mountedPlayersForm) {
      mountedPlayersForm = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));

    }
    return mountedPlayersForm;
  };

  beforeEach(function () {
    props = {
      onStartGame: jest.fn(),
      players: [],
      result: { status: 'waiting' },
      turn: '' };

    mountedPlayersForm = undefined;
  });

  it('should render the PlayersForm', function () {
    var playersFormWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));
    expect(playersFormWrapper).toMatchSnapshot();
  });

  // it('should render a form as root element', () => {
  //   const playersFormElement = playersForm().find('form');
  //   let wrapping = playersFormElement.first();
  //   expect(wrapping.children()).toEqual(playersForm().children());
  // });

  describe('legend', function () {
    it('should render a legend inside the form if the result.status is not running', function () {

    });

    it('should not render a legend inside the form if the result.status is running', function () {

    });
  });

  describe('div.player-input', function () {
    it('should render two div.render-input', function () {

    });

    it('');
  });

  it('should render two label', function () {

  });

  describe('input type=text', function () {
    it('should render two input type=text', function () {

    });

    it('simulate change event', function () {

    });
  });

  describe('turn', function () {
    it('should render the div.turn inside the form if the result.status is running', function () {

    });

    it('should not render div.turn inside the form if the result.status is not running', function () {

    });
  });

  describe('start game button', function () {
    it('should render the button.start-game-button inside the form if the result.status is not running', function () {

    });

    it('should not render the button.start-game-button inside the form if the result.status is running', function () {

    });

    it('simulate click event', function () {

    });
  });
});