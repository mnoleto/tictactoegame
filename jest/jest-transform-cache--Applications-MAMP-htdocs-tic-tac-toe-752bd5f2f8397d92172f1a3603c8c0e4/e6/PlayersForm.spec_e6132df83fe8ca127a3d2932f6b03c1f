/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_PlayersForm = require('./PlayersForm'); /* istanbul ignore next */var _PlayersForm2 = _interopRequireDefault(_PlayersForm);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayersForm is responsible for render the game grid
// CONTRACTS
// 1. 

describe('PlayersForm', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayersForm = /* istanbul ignore next */void 0;
  var playersForm = function playersForm() {
    if (!mountedPlayersForm) {
      mountedPlayersForm = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayersForm2.default, props));

    }
    return mountedPlayersForm;
  };

  beforeEach(function () {
    props = {};
    mountedPlayersForm = undefined;
  });

  it('always render the PlayersForm', function () {
    var playersFormElement = playersForm().find('div');
    expect(playersFormElement.length).toEqual(1);
  });
});