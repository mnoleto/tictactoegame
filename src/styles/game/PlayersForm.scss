@import '../_colors.scss';

$input-height: 50px;

.player-input {
  display: block;
  margin: 0px 5px 10px;
  // width: 50%;
  label {
    background-color: #fff;
    border-radius: 4px 0px 0px 4px;
    color: $blue;
    display: inline-block;
    height: $input-height;
    font-family: 'Quicksand', sans-serif;
    font-size: 200%;
    font-weight: 700;
    line-height: $input-height;
    padding: 0px 10px;
    width: 15%;
  }
  input[type="text"] {
    border: 0px;
    border-radius: 0px 4px 4px 0px;
    color: $dark-blue;
    display: inline-block;
    font-size: 200%;
    height: $input-height;
    width: 85%;
  }
  &.inactive label, &.inactive input {
    opacity: 0.5;
  }
  @media (min-width: 992px) {
    display: inline-block;
    width: auto;
  }
}

.players-form {
  margin-bottom: 50px;
  width: 100%;
  text-align: center;
  legend {
    font-size: 80%;
    letter-spacing: 1px;
    margin-bottom: 10px;
    text-align: center;
    text-transform: uppercase;
    width: 100%;
    @media (min-width: 992px) {
      font-size: 120%;
    }
  }
}

.start-game-button {
  background-color: lighten($blue, 20%);
  border: 0px;
  color: $dark-blue;
  display: block;
  font-size: 100%;
  margin: 15px auto 10px;
  padding: 10px 20px;
}
