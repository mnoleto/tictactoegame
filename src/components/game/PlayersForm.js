import React, { Component, PropTypes } from 'react';
import PlayerInput from './PlayerInput';
import '../../styles/game/PlayersForm.scss';

// PlayersForm is responsible for render the players form
class PlayersForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      playerX: '',
      playerO: ''
    };

    this.handleInputChange = this.handleInputChange.bind(this);
    this.renderFormFooter = this.renderFormFooter.bind(this);
    this.renderFormHeader = this.renderFormHeader.bind(this);
    this.sendData = this.sendData.bind(this);
  }

  componentWillReceiveProps() {
    if(this.props.players.length > 0 && this.props.players[0] !== this.state.playerX  && this.props.players[1] !== this.state.playerO ) {
      this.setState({
        playerX: this.props.players[0],
        playerO: this.props.players[1]
      });
    }
  }

  handleInputChange(name, value) {
    this.setState({[name]: value});
  }

  sendData() {
    const { onStartGame } = this.props;
    let playerX = this.state.playerX,
      playerO = this.state.playerO;

    onStartGame(playerX, playerO);
  }

  renderFormFooter() {
    const { result, turn } = this.props;
    if(result.status && (result.status && result.status === 'running')) {
      return (
        <div className="turn">It`s <strong>{turn}</strong> turn.</div>
      );
    } else if(!result.status || (result.status && result.status !== 'running')) {
      return (
        <button type="button" className="start-game-button" onClick={this.sendData}>start game</button>
      );
    }
  }

  renderFormHeader() {
    const { result } = this.props;
    if(!result.status || (result.status && result.status !== 'running')) {
      return (
        <legend>Please, inform the players name below:</legend>
      );
    }
    return '';
  }

  render() {
    const { result, turn } = this.props;
    
    return (
      <form className="players-form">
        {this.renderFormHeader()}

        <PlayerInput
          label="X"
          onInputChange={this.handleInputChange}
          player="playerX"
          result={result}
          turn={turn}
          value={this.state.playerX || ''} />

        <PlayerInput
          label="O"
          onInputChange={this.handleInputChange}
          player="playerO"
          result={result}
          turn={turn}
          value={this.state.playerO || ''} />

        {this.renderFormFooter()}
      </form>
    );
  }
}

PlayersForm.propTypes = {
  onStartGame: PropTypes.func.isRequired,
  players: PropTypes.array,
  result: PropTypes.object.isRequired,
  turn: PropTypes.string
};

export default PlayersForm;