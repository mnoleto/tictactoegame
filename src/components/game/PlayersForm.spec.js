import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import PlayersForm from './PlayersForm';

// PlayersForm is responsible for render the game grid
describe('PlayersForm', () => {
  let props, mountedPlayersForm;
  const playersForm = () => {
    if(!mountedPlayersForm) {
      mountedPlayersForm = mount(
        <PlayersForm {...props} />
      );
    }
    return mountedPlayersForm;
  };

  beforeEach(() => {
    props = {
      onStartGame: jest.fn(),
      players: [],
      result: {status: 'waiting'},
      turn: ''
    };
    mountedPlayersForm = undefined;
  });

  it('should render the PlayersForm', () => {
    let playersFormWrapper = shallow(<PlayersForm {...props} />);
    expect(playersFormWrapper).toMatchSnapshot();
  });

  it('form contains everything else that gets rendered', () => {
    let playersFormElement = playersForm().find('form');
    let wrapping = playersFormElement.first();
    expect(wrapping.children()).toEqual(playersForm().children());
  });

  describe('legend', () => {
    it('should render a legend inside the form if the result.status is not running', () => {
      let playersFormRenderer = playersForm();
      playersFormRenderer.setProps({result: {status: 'waiting'}});
      let legendElement = playersForm().find('legend');
      expect(legendElement.length).toEqual(1);
    });

    it('should not render a legend inside the form if the result.status is running', () => {
      let playersFormRenderer = playersForm();
      playersFormRenderer.setProps({result: {status: 'running'}});
      let legendElement = playersForm().find('legend');
      expect(legendElement.length).toEqual(0);
    });
  });
  
  describe('div.player-input', () => {
    it('should render two div.player-input', () => {
      let inputElements = playersForm().find('div.player-input');
      expect(inputElements.length).toEqual(2);
    });
  });

  describe('start game button', () => {
    it('should render the button.start-game-button inside the form if the result.status is not running', () => {
      let playersFormRenderer = playersForm();
      playersFormRenderer.setProps({result: {status: 'waiting'}});
      let buttonElement = playersForm().find('button');
      expect(buttonElement.length).toEqual(1);
    });

    it('should not render the button.start-game-button inside the form if the result.status is running', () => {
      let playersFormRenderer = playersForm();
      playersFormRenderer.setProps({result: {status: 'running'}});
      let buttonElement = playersForm().find('button');
      expect(buttonElement.length).toEqual(0);
    });

    it('simulate click event', () => {
      props.onStartGame = sinon.spy();
      let buttonElement = playersForm().find('button');

      buttonElement.simulate('click');
      expect(props.onStartGame.callCount).toEqual(1);
    });
  });
});