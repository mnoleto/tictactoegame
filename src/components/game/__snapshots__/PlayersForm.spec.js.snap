exports[`PlayersForm should render the PlayersForm 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    className="players-form">
    <legend>
        Please, inform the players name below:
    </legend>
    <PlayerInput
        label="X"
        onInputChange={[Function]}
        player="playerX"
        result={
            Object {
                "status": "waiting",
              }
        }
        turn=""
        value="" />
    <PlayerInput
        label="O"
        onInputChange={[Function]}
        player="playerO"
        result={
            Object {
                "status": "waiting",
              }
        }
        turn=""
        value="" />
    <button
        className="start-game-button"
        onClick={[Function]}
        type="button">
        start game
    </button>
</form>,
  "nodes": Array [
    <form
      className="players-form">
      <legend>
            Please, inform the players name below:
      </legend>
      <PlayerInput
            label="X"
            onInputChange={[Function]}
            player="playerX"
            result={
                  Object {
                        "status": "waiting",
                      }
            }
            turn=""
            value="" />
      <PlayerInput
            label="O"
            onInputChange={[Function]}
            player="playerO"
            result={
                  Object {
                        "status": "waiting",
                      }
            }
            turn=""
            value="" />
      <button
            className="start-game-button"
            onClick={[Function]}
            type="button">
            start game
      </button>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PlayersForm
        onStartGame={[Function]}
        players={Array []}
        result={
                Object {
                        "status": "waiting",
                      }
        }
        turn="" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PlayersForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleInputChange": [Function],
        "props": Object {
          "onStartGame": [Function],
          "players": Array [],
          "result": Object {
            "status": "waiting",
          },
          "turn": "",
        },
        "refs": Object {},
        "renderFormFooter": [Function],
        "renderFormHeader": [Function],
        "sendData": [Function],
        "state": Object {
          "playerO": "",
          "playerX": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          className="players-form">
          <legend>
                    Please, inform the players name below:
          </legend>
          <PlayerInput
                    label="X"
                    onInputChange={[Function]}
                    player="playerX"
                    result={
                              Object {
                                        "status": "waiting",
                                      }
                    }
                    turn=""
                    value="" />
          <PlayerInput
                    label="O"
                    onInputChange={[Function]}
                    player="playerO"
                    result={
                              Object {
                                        "status": "waiting",
                                      }
                    }
                    turn=""
                    value="" />
          <button
                    className="start-game-button"
                    onClick={[Function]}
                    type="button">
                    start game
          </button>
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          className="players-form">
          <legend>
                    Please, inform the players name below:
          </legend>
          <PlayerInput
                    label="X"
                    onInputChange={[Function]}
                    player="playerX"
                    result={
                              Object {
                                        "status": "waiting",
                                      }
                    }
                    turn=""
                    value="" />
          <PlayerInput
                    label="O"
                    onInputChange={[Function]}
                    player="playerO"
                    result={
                              Object {
                                        "status": "waiting",
                                      }
                    }
                    turn=""
                    value="" />
          <button
                    className="start-game-button"
                    onClick={[Function]}
                    type="button">
                    start game
          </button>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PlayersForm
    onStartGame={[Function]}
    players={Array []}
    result={
        Object {
            "status": "waiting",
          }
    }
    turn="" />,
}
`;
